haystack_experimental/dataclasses/chat_message.py

```diff
@@ -35,9 +35,9 @@ class ToolCall:
     :param arguments: The arguments to call the Tool with.
     """
 
-    id: Optional[str]  # noqa: A003
     tool_name: str
     arguments: Dict[str, Any]
+    id: Optional[str] = None  # noqa: A003
```

Comment:
unrelated change.

Previously, I forgot to provide a default for `id`.
Dataclasses require attributes with defaults to be placed after others.